Togglable inputs; toggle_duck, toggle_jump, toggle_sprint, toggle_walk

braem opened the pull request:
Closes #921
Closes #923

- Adds `toggle_jump`, `toggle_sprint`, and `toggle_walk` commands. 
- Unlocks `toggle_duck` from being ahop only.
- Locks `toggle_jump` to gamemodes with autohop
- Locks `toggle_sprint` and `toggle_walk` for gamemodes that have those (using `GameModeHUDCapability`)
- Moved `toggle_duck` into `mom_player` to better centralize the togglable inputs.
- Overrides `PlayerRunCommand`, appending inputs if they have been toggled on.
- Connected with the hud keypress display. Have to network all toggles besides duck for more accurate display.
- `toggle_sprint` and `toggle_walk` untoggle when player is stopped from sprinting/walking.
- Removed unused code from `player` and `in_main` & minorly fixed up the keypress display (`OVERRIDE` -> `override`, moving multi-line method out of class declaration, changed bgcolor to be an animation var instead of hardcoded).
- Added to keybinds:
![image](https://user-images.githubusercontent.com/9014762/92431818-4f187480-f14d-11ea-9459-94485350e96e.png)


### Checklist
- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

Gocnak commented on mp/src/game/client/momentum/c_mom_player.h:
```diff
@@ -52,6 +52,9 @@ class C_MomentumPlayer : public C_BasePlayer, public CMomRunEntity
     CNetworkVar(int, m_iDirection); // Used in kickback effects for player
     CNetworkVar(int, m_iLastZoomFOV); // Last FOV when zooming
     CNetworkVar(bool, m_bSurfing);
+    CNetworkVar(bool, m_bJumpToggled); // Used by keypress hud
+    CNetworkVar(bool, m_bSprintToggled); // Used by keypress hud
+    CNetworkVar(bool, m_bWalkToggled);   // Used by keypress hud
```
I feel like this can be simplified with a `CNetworkVar(int, m_iButtonsToggled)`, and re-using the IN_SPRINT / WALK / etc. Saves the need for 3 separate bools.

braem commented:
Did the proposed change, nicely simplifies things. Also added a check for duck toggled in keypress hud to properly display that; captures the time between pressing and fully ducking.

Gocnak commented on mp/src/game/shared/momentum/mom_system_gamemode.cpp:
```diff
@@ -60,6 +61,11 @@ void CGameModeBase::OnPlayerSpawn(CMomentumPlayer *pPlayer)
+    if (pPlayer->m_iButtonsToggled & IN_DUCK)
+        pPlayer->m_iButtonsToggled &= ~IN_DUCK;
+    if (pPlayer->m_iButtonsToggled & IN_JUMP)
+        pPlayer->m_iButtonsToggled &= ~IN_JUMP;
```
All toggled buttons should just reset on player spawn, regardless of mode. They should also reset upon mom_restart commands

Gocnak commented on mp/src/game/server/momentum/mom_player.cpp:
```diff
@@ -2116,18 +2144,60 @@ void CMomentumPlayer::DoPaint()
+    if (pPlayer->m_iButtonsToggled & IN_WALK)
+        pPlayer->m_iButtonsToggled &= ~IN_WALK;
+    else
+        pPlayer->m_iButtonsToggled |= IN_WALK;
```
This is repeated multiple times, you can pull this out to a public `ToggleInput` void that takes the input flag as a param

Gocnak commented on mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp:
```diff
@@ -39,26 +39,22 @@ class CHudKeyPressDisplay : public CHudElement, public Panel
     CPanelAnimationVar(Color, m_Normal, "KeyPressedColor", "Panel.Fg");
     CPanelAnimationVar(Color, m_darkGray, "KeyOutlineColor", "Dark Gray");
     CPanelAnimationVar(Color, m_Disabled, "KeyDisabledColor", "Red");
+    CPanelAnimationVar(Color, m_BgColor, "BgColor", "0 0 0 1"); // empty background, 1 alpha so game text doesn't obscure our text
```
I'm not even sure what the original intent was with the 1 alpha, but if the background color needs set there's `BgColor_override` inside every panel

braem replied:
Not sure why either, works just fine with 0 alpha.

braem commented:
Implemented the requested changes. Also I found that the run safeguards are hit when inputs are toggled on, should this be the case? For now I changed it to only safeguard if the button is held down, not toggled.

Gocnak commented:
I'm inclined to say yes; it's still movement keys, and they took the easy way out of toggling it instead of holding it, so I would expect it to behave as if they have it pressed (as does the `hud_keypress` panel).

Gocnak commented on mp/src/game/shared/momentum/run/mom_run_safeguards.cpp:
```diff
@@ -84,6 +84,7 @@ bool CRunSafeguard::IsSafeguarded(RunSafeguardMode_t mode)
 #else
     pPlayer->m_nButtons;
 #endif
+    nButtons &= ~pPlayer->m_nButtonsToggled;
```
See https://github.com/momentum-mod/game/pull/929#issuecomment-692444896
